### PR TITLE
Initialization output cleanup - eio output and table output

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
Addresses #6001.

Restores the construction name in the surface details output and cleans up some issues with the new (in v8.7) InitializationSummary table output. 

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by mjwitte** [comment]
And this should eliminate the `KeyError: /#Shading Surfaces` CI regression test failures.

**Comment by mjwitte** [comment]
@Myoldmopar @mbadams5 Wanted to unit test eio output, so the latest commit here starts a partial implementation of eso_stream.  Is it ok to leave a mix of writes to eso_stream and gio::write for the same file?  And just convert more to stream as needed.

File: src/EnergyPlus/OutputReports.cc [hunk 1]
```diff
@@ -1987,52 +1983,67 @@ DetailsForSurfaces( int const RptType ) // (1=Vertices only, 10=Details only, 11
 					ExtConvCoeffCalc = ConvCoeffCalcs( std::abs( Surface( surf ).ExtConvCoeff ) );
 				}
 				if ( Surface( surf ).ExtBoundCond == ExternalEnvironment ) {
-					{ IOFlags flags; flags.ADVANCE( "No" ); gio::write( unit, Format_705, flags ) << "ExternalEnvironment"; }
-					{ IOFlags flags; flags.ADVANCE( "No" ); gio::write( unit, Format_705, flags ) << ExtConvCoeffCalc; }
-					{ IOFlags flags; flags.ADVANCE( "No" ); gio::write( unit, Format_705, flags ) << IntConvCoeffCalc; }
+					if ( DataGlobals::eio_stream ) *DataGlobals::eio_stream << "ExternalEnvironment" << "," << ExtConvCoeffCalc << "," << IntConvCoeffCalc << ",";
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutputReports.cc]: :+1: to all these changes.  The only question I'd bring up is whether we actually want to protect every single call with an if check.  Or if we can collect the `if` check to a more centralized location and abort early if it is null.  Not necessary to change now, but if we're already in here cleaning out the commented lines, then maybe it's worth discussion.

File: src/EnergyPlus/OutputReports.cc [hunk 2]
```diff
@@ -1738,16 +1738,15 @@ DetailsForSurfaces( int const RptType ) // (1=Vertices only, 10=Details only, 11
 	std::string AlgoName;
 
 	// Formats
-	static gio::Fmt Format_700( "('! <Zone Surfaces>,<Zone Name>/#Shading Surfaces,# Surfaces')" );
-	static gio::Fmt Format_700b( "('! <Shading Surfaces>,<Zone Name>/#Shading Surfaces,# Surfaces')" );
+	static gio::Fmt Format_700( "('! <Zone Surfaces>,Zone Name,# Surfaces')" );
+	static gio::Fmt Format_700b( "('! <Shading Surfaces>,Number of Shading Surfaces,# Surfaces')" );
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutputReports.cc]: Yay!!

**Review by Myoldmopar — CHANGES_REQUESTED** [review]
This all looks good.  I'll leave it as "request changes" since, at a minimum, we need to clean out the code you left in as comments.  I cannot find anything wrong with your changes and appreciate you going through the effort to add the eio stream here.  If @mbadams5 has any comments, since the unit test structure here is more of his baby, they'd far outweigh my opinion.  But if he doesn't have any issues, then 👍

**Comment by mbadams5** [comment]
The unit test and fixture side looks good.

You might need to update unit tests in the future as you convert more gio::write to the stream but that is the only thing I can see.

**Comment by mjwitte** [comment]
@Myoldmopar Forgot to merge in develop, and also checking out possible issue with reporting to tubular daylighting devices.  So, not quite ready yet.

**Comment by mjwitte** [comment]
@Myoldmopar This should be ready for review now.
@mbadams5 Thanks for looking at this.

**Pull request merged**